### PR TITLE
SavingForTheFuture

### LINKED ISSUE #45: Saving data for future
We should not be able to save data for future. The last acceptable record is usually of the previous day.

### PR DESCRIPTION
**Pull request by Patowhiz**
fixes #45 Saving data for future .
@africanmathsinitiative/developers this is ready for review.
@dannyparsons please review this so that we can move forward with implementing the same for the other forms.

### CODE REVIEW

File: ClimsoftVer4/ClimsoftVer4/ucrHourlyWind.vb [hunk 1]
```diff
@@ -330,8 +324,44 @@ Public Class ucrHourlyWind
...
+        ucrLinkedYear = ucrYearControl
+        ucrLinkedMonth = ucrMonthControl
+        ucrLinkedDay = ucrDayControl
+    End Sub
+
+    ''' <summary>
+    ''' checks the selected year month day to permit entry or not.
+    ''' this prevents data entry of current and future dates
+    ''' </summary>
+    Private Sub ValidateDataEntryPermission()
```
Comment (Patowhiz) [inline ClimsoftVer4/ClimsoftVer4/ucrHourlyWind.vb:359]: I'm not sure if this is the right name for this subroutine

**Comment by Patowhiz** [comment]
The main addition here is

```
''' <summary>
    ''' checks the selected year month day to permit entry or not.
    ''' this prevents data entry of current and future dates
    ''' </summary>
    Private Sub ValidateDataEntryPermission()
        'if its an update or any of the linked year,month and day selector is nothing then just exit the sub
        If bUpdating OrElse ucrLinkedYear Is Nothing OrElse ucrLinkedMonth Is Nothing OrElse ucrLinkedDay Is Nothing Then
            Exit Sub
        End If

        Dim todayDate As Date
        Dim selectedDate As Date

        'initialise the dates with ONLY year month and day values. 
        'Neglect the time factor
        todayDate = New Date(DateTime.Now.Year, DateTime.Now.Month, DateTime.Now.Day)
        selectedDate = New Date(ucrLinkedYear.GetValue, ucrLinkedMonth.GetValue, ucrLinkedDay.GetValue)

        'if selectedDate is earlier than todayDate enable control
        If DateTime.Compare(selectedDate, todayDate) < 0 Then
            Me.Enabled = True
        Else
            'if it is same time (0) or later than (>0) disable control
            Me.Enabled = False
        End If
    End Sub
```

I chose to disable the entire control instead of the individual controls to preserve any enable properties already set for the individual textboxes.
This sub is called in` PopulateControl()`

**Comment by maxwellfundi** [comment]
@Patowhiz  is this ready?

**Comment by Patowhiz** [comment]
That's ready

**Pull request merged**